File: src/TinRoll.Data/TinRollContext.cs
﻿using Microsoft.EntityFrameworkCore;
using TinRoll.Data.Entities;

namespace TinRoll.Data
{
    public class TinRollContext : DbContext
    {
        public TinRollContext(DbContextOptions<TinRollContext> options) : base(options)
        { }

        public DbSet<User> Users { get; set; }

        public DbSet<Question> Questions { get; set; }

        //public Task<int> SaveChanges()
        //{
        //    ChangeTracker.Entries().Where(E => E.State == EntityState.Added).ToList(); 
        //}
    }
}

File: src/api/TinRoll.Data/TinRollContext.fs
namespace TinRoll.Data

open Microsoft.EntityFrameworkCore

type TinRollContext(options : DbContextOptions<TinRollContext>) =
    inherit DbContext(options)

    [<DefaultValue>]
    val mutable questions:DbSet<Question>
    member x.Questions
        with get() = x.questions
        and set v = x.questions <- v

   